### PR TITLE
chore(test): Add end-to-end test suite

### PR DESCRIPTION
**Pull request by charithe**
This PR adds a set of E2E tests that verify the Cerbos deployment process and exercises the different storage backends. I discovered two bugs whilst working on it so the work has already paid off in some way.

This is purely foundational work with lots of room for improvement in the future. However, it should be good enough for our immediate requirements of increasing the confidence on new releases.

CI workflow will be added in a separate PR.

Part of #147

### CODE REVIEW

**Comment by charithe** [comment]
Proto check failure is due to `buf.lock` in `main` having references to versions that seem to have been purged from BSR. It should  be fixed after the updated `buf.lock` file in this PR is merged.

File: e2e/blob/helmfile.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      - auth:
+          rootUser: admin
+          rootPassword: passw0rd
+      - defaultBuckets: "cerbos"
+      - persistence:
+          enabled": false
+
+  - name: cerbos
+    namespace: '{{ requiredEnv "E2E_NS" }}'
+    needs: ["postgres"]
```
Comment (dbuduev) [inline e2e/blob/helmfile.yaml:46]:
```suggestion
    needs: ["minio"]
```

**Review by dbuduev — APPROVED** [review]
LGTM 👍🏻 

> telepresence: error: connector.Connect: failed to start traffic manager: the helm operation timed out.

I ran into this problem though. It might be due to Apple silicon.

**Comment by charithe** [comment]
I ran into that problem too. Had to increase the timeout:

```sh
cat ~/.config/telepresence/config.yml 
timeouts:
  helm: 60s
```

**Comment by dbuduev** [comment]
Thank you! All good now.

I had tried to increase the timeout but misplaced the config.

**Pull request merged**